spec/marc_to_solr/lib/config_spec.rb

```diff
@@ -170,22 +174,35 @@ def fixture_record(fixture_name)
   end
   describe "call_number_display field" do
     it "returns the call_number_display field with k subfield in the beginning" do
-      record = @indexer.map_record(fixture_record('9957270023506421'))
-      expect(record['call_number_display']).to eq(["Eng 20Q 6819 "])
+      expect(@record_call_number1['call_number_display']).to eq(["Eng 20Q 6819"])
+    end
+    it "skips indexing the field if subfields $h and $i and $k are missing" do
+      expect(@record_call_number_nil['call_number_display']).to be nil
+    end
+    it "doesnt have trailing spaces" do
+      expect(@record_call_number2['call_number_display']).to eq(["CD- 50000"])
     end
   end
 
   describe "call_number_browse field" do
-    it "returns the call_number_browse field with k subfield at the end" do
-      record = @indexer.map_record(fixture_record('9957270023506421'))
-      expect(record['call_number_browse_s']).to eq(["6819 Eng 20Q"])
+    it "indexes the call_number_browse field" do
+      expect(@sample40['call_number_browse_s']).to eq([" .B7544 2003q Oversize RA566.27"])
```

Comment:
@mzelesky for mms_id 9941598513506421 we have :
`<datafield tag="852" ind1="0" ind2=" ">
    <subfield code="b">firestone</subfield>
    <subfield code="c">stacks</subfield>
    <subfield code="h"/>
    <subfield code="i">.B7544 2003q</subfield>
    <subfield code="k">Oversize RA566.27</subfield>
    <subfield code="8">22172120500006421</subfield>
  </datafield>`

h is an empty space in front. Do I keep this empty space and then I would have call_number -> `" .B7544 2003q Oversize RA566.27"` or do I remove the leading spaces as well and then the call_number would be -> `".B7544 2003q Oversize RA566.27"` with no space in front?

Reply:
In current production we have these spaces. https://catalog.princeton.edu/catalog/4159851/raw
But I think we should't. 
I will update the PR to remove the spaces.

Reply:
@mzelesky I dont know if you have time to take a look but I removed all the spaces from empty subfields for the call_number fields. Also for the call number fields in the holdings_1display.